// helper_modules/clientEmail.js
var nodemailer = require("nodemailer");
var crypto = require("crypto");

function sendEmail(email,name,firstname){

    var id = name;
    var hash = crypto.createHash('sha256').update(name).digest('hex');

    var smtpTransport = nodemailer.createTransport("SMTP",{
	host: 'mail.gandi.net',
	port: 587,
	use_authentication: true,
	user: "starters@starters-singapore.com",
	pass: "deutschebank"
    });

    smtpTransport.sendMail({
	from: "Geoffrey & Jae <starters@starters-singapore.com>",
	to: email,
	subject: "Kick off your Starter",
	text: "Hi " + firstname + ",\n\nThanks for signing up with us for a Starter. We're searching high and low for people who we think you and your friends will enjoy kicking the night off with, so please bear with us while we work our mojo. Remember that you need to bring TWO of your best buddies in order to get matched. We posted the invite link below for you to get started. We added a couple extra invites too, in case you needed some backup ;) \n\n\nwww.starters-singapore.com/invite?name=" + name.replace(' ', '%20') + "&id=" + hash + "\n\n\nWe'll contact you as soon as everything is set up. Thanks for signing up " + firstname + ", and please feel free to reach out to us personally if there's anything on your mind. Enjoy! \n\nRegards,\nGeoffrey Tsui\nStarters Singapore\ngeoffrey@starters-singapore.com\n+65 91759342\n\nJaeYong Heo\nStarters Singapore\njae@starters-singapore.com\n+65 98323313",
    }, function(error, response){
	if(error){
	   console.log(error);
	} else{
	    console.log("Success!");
	}
    });
}

exports.sendEmail = sendEmail;
