// helper_modules/inviteEmail.js
var nodemailer = require("nodemailer");
var crypto = require("crypto");


function sendEmail(email,name, host){

    var hash = crypto.createHash('sha256').update(host).digest('hex');

    var smtpTransport = nodemailer.createTransport("SMTP",{
	host: 'mail.gandi.net',
	port: 587,
	use_authentication: true,
	user: "geoffrey@starters-singapore.com",
	pass: "deutschebank"
    });

    smtpTransport.sendMail({
	from: "Geoffrey at Starters <geoffrey@starters-singapore.com>",
	to: email,
	subject: "Kick off your Starter",
	text: "Hi " + name + ",\n\n" + host + " has invited you to join their Starter. If you're not familiar, Starters is a social club that aims to bring two groups of interesting people together: one group of three boys and one group of three girls, just to spice things up ;).\nYou can sign up on our site below (Make sure to fill in your friend's name as your referral!): \n\nwwww.starters-singapore.com\n\nPlease don't hesitate to contact me personally if you have any questions or feedback. We look forward to hosting your Starter!\n\nRegards,\nGeoffrey Tsui\nStarters Singapore\ngeoffrey@starters-singapore.com\n+65 91759342",
    }, function(error, response){
	if(error){
	   console.log(error);
	} else{
	    console.log("Success!");
	}
    });
}

exports.sendEmail = sendEmail;
